### ISSUE TITLE
Slow API performance when creating clusters

### ISSUE THREAD

**Issue by paynejacob** [rancher/rancher#36249]
**Rancher Server Setup**
- Rancher version: 2.6.2
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): helm

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: admin

**Describe the bug**
Listing resources with the v1 api can become slow when a cluster is being created or deleted.

**To Reproduce**
1. have a rancher ha install on 2 cpu with 8g ram.
2. create a cluster
3. observe the api response times increase while the cluster is provisioning

**Result**
api response times increase while the cluster is provisioning

SURE-3815

**Comment by paynejacob** [comment]
I tested on a 3 node helm install of rancher 2.6.2 and enabled monitoring. Each node had 8gb of ram.
`corral create ghcr.io/paynejacob/corral/sure-3815:latest`

I pulled 3 endpoints in a loop as a benchmark

1. `/v1/rke-machine.cattle.io.amazonec2machinetemplates` as it was mentioned in the issue
2. `/v1/management.cattle.io.nodes` this is a resource I expected to similar activity on as 1.
3. `/v1/ui.cattle.io.navlinks` this was a control of sorts to see if the performance issues could affect any endpoint in rancher.

I ran the following test:

1. create the test cluster
2. curl the endpoints above in a tight loop sequentially and log their response times
3. create 10 ec2 clusters in a tight loop sequentially with 1 node pool each (10 was selected because it was when I first saw differences)
4. wait for all 10 clusters to be ready and let them idle for ~10m
5. delete all the clusters at once.

I then repeated this test without step 2 to see if the constant api activity was related to the performance issues.

I found that all endpoints, not just the `amazonec2machinetemplates` endpoint, slowed down when clusters were being create and more significantly when clusters were being deleted.  At it's worst all 3 endpoints were responding in ~12s.  Watching Grafana I observed all 3 rancher pods memory increase by ~50% during the cluster creates and ~ 100% during cluster deletes.  CPU increased as well at these times. Given the slow down occurs on all 3 endpoints I think the issues is likely related to when write or delete operations occur while and endpoint is gathering results.  Serializing data is already know to be a performance issue that can dramatically increase cpu and memory load.  It is possible the deletes are non performance and this makes the api issues more apparent due to 2 non performance activities happening at the same time or the deletes are somehow exacerbating the api issues.  I do not see anything unique about the `amazonec2machinetemplates` results, structure, or any code in rancher that would lead me to believe this problem is related to that specific resource.   I think this issues is going to be resolved by broader performance improvements around listing api results and deleting many resources.  

I additionally wanted to mention when I was initially testing I used a cluster with only 4g of memory per node.  With monitoring installed these issues quickly took down my cluster and I was seeing response times increase more dramatically (50s +) before the cluster ultimately crashed.  It is possible that there are resource constraints in the hosted rancher setup that make this issue more apparent.
